go/cmd/vtexplain/vtexplain.go

```diff
@@ -29,62 +28,46 @@ import (
 	"vitess.io/vitess/go/vt/servenv"
 	"vitess.io/vitess/go/vt/vtexplain"
 
-	querypb "vitess.io/vitess/go/vt/proto/query"
+	"github.com/spf13/pflag"
 
-	// Include deprecation warnings for soon-to-be-unsupported flag invocations.
-	_flag "vitess.io/vitess/go/internal/flag"
+	querypb "vitess.io/vitess/go/vt/proto/query"
 )
 
 var (
-	sqlFlag            = flag.String("sql", "", "A list of semicolon-delimited SQL commands to analyze")
-	sqlFileFlag        = flag.String("sql-file", "", "Identifies the file that contains the SQL commands to analyze")
-	schemaFlag         = flag.String("schema", "", "The SQL table schema")
-	schemaFileFlag     = flag.String("schema-file", "", "Identifies the file that contains the SQL table schema")
-	vschemaFlag        = flag.String("vschema", "", "Identifies the VTGate routing schema")
-	vschemaFileFlag    = flag.String("vschema-file", "", "Identifies the VTGate routing schema file")
-	ksShardMapFlag     = flag.String("ks-shard-map", "", "JSON map of keyspace name -> shard name -> ShardReference object. The inner map is the same as the output of FindAllShardsInKeyspace")
-	ksShardMapFileFlag = flag.String("ks-shard-map-file", "", "File containing json blob of keyspace name -> shard name -> ShardReference object")
-	numShards          = flag.Int("shards", 2, "Number of shards per keyspace. Passing --ks-shard-map/--ks-shard-map-file causes this flag to be ignored.")
-	executionMode      = flag.String("execution-mode", "multi", "The execution mode to simulate -- must be set to multi, legacy-autocommit, or twopc")
-	replicationMode    = flag.String("replication-mode", "ROW", "The replication mode to simulate -- must be set to either ROW or STATEMENT")
-	normalize          = flag.Bool("normalize", false, "Whether to enable vtgate normalization")
-	outputMode         = flag.String("output-mode", "text", "Output in human-friendly text or json")
-	dbName             = flag.String("dbname", "", "Optional database target to override normal routing")
-	plannerVersionStr  = flag.String("planner-version", "", "Sets the query planner version to use when generating the explain output. Valid values are V3 and Gen4")
-
-	// vtexplainFlags lists all the flags that should show in usage
-	vtexplainFlags = []string{
-		"output-mode",
-		"planner-version",
-		"normalize",
-		"shards",
-		"replication-mode",
-		"schema",
-		"schema-file",
-		"sql",
-		"sql-file",
-		"vschema",
-		"vschema-file",
-		"ks-shard-map",
-		"ks-shard-map-file",
-		"dbname",
-		"queryserver-config-passthrough-dmls",
-	}
+	sqlFlag            string
+	sqlFileFlag        string
+	schemaFlag         string
+	schemaFileFlag     string
+	vschemaFlag        string
+	vschemaFileFlag    string
+	ksShardMapFlag     string
+	ksShardMapFileFlag string
+	numShards          int
```

Comment:
this used to default to 2; the pattern we have been doing when the default is not the zero-value for the given type is:

```go
var myFlag = 2


func registerFlags(fs *pflag.FlagSet) {
    fs.IntVar(&myFlag, "my-flag", myFlag, "my flag's usage text")
}
```

Reply:
(and then ditto for the `string` variables that don't default to the empty string, and so on)

Reply:
til